### ISSUE TITLE
URL and Custom Filter regex on target

### ISSUE THREAD

**Issue by johngodley** [johngodley/redirection#1220]
> I’m trying to use the custom filter feature to redirect only on non-admin pages. My source URL is (.*)\.php$ with the Regex box checked (this is working). But the Matched Target field doesn’t want to regard the regex capture group.
URL: (.*)\.php$
Matched Target: $1
Result is: site.com/myurl.php then goes to site.com/1

> This pairing works fine as a regular redirect, but when using “URL and Custom Filter”, the Matched Target doesn’t seem to be regex-ready with the capture group. Am I missing something?

From https://wordpress.org/support/topic/url-and-custom-filter-regex-on-target/

**Comment by jdeglopper** [comment]
Hi John, I am seeing this issue as well when trying to use a regex match along with a URL and Server filter.  

redirection 3.5, Wordpress 4.9.5, PHP 7.1.22, Apache 2.4.6 (CentOS 7).

```
jacob@nitrogen ~ $ curl -v http://johndoe.com/rates/   
*   Trying 168.143.xxx.yyy...
* TCP_NODELAY set
* Connected to johndoe.com (168.143.xxx.yyy) port 80 (#0)   
> GET /rates/ HTTP/1.1
> Host: johndoe.com   
> User-Agent: curl/7.55.1
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Date: Thu, 25 Oct 2018 21:28:54 GMT
< Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.1.22
< X-Powered-By: PHP/7.1.22
< Expires: Wed, 11 Jan 1984 05:00:00 GMT
< Cache-Control: no-cache, must-revalidate, max-age=0
< Location: https://example.com/officers/john-doe/1/?c=true       
< X-Redirect-Agent: redirection
< Vary: User-Agent
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
<
```
The database entry for this redirect looks like this:

```
*************************** 1. row ***************************
         id: 757
        url: ^/(.*)
      regex: 1
   position: 1
 last_count: 4
last_access: 2018-10-25 17:28:54
   group_id: 1
     status: enabled
action_type: url
action_code: 301
action_data: a:3:{s:8:"url_from";s:49:"https://example.com/officers/john-doe/$1/?c=true";s:11:"url_notfrom";s:0:"";s:6:"server";s:21:"http://johndoe.com";}      
 match_type: server
      title:
1 row in set (0.00 sec)
```

In my case, both johndoe.com and example.com are pointed to the same Wordpress install to allow vanity pages to direct to an interior page on the canonical URL, replaced here by example.com.  

wp-admin or SSH access is available.

**Comment by jdeglopper** [comment]
Ah, I see this is #1162 and not currently supported.  I hope it's something that can be added soon!

**Comment by johngodley** [comment]
Server regex matching was fixed in #1232, and custom filter in #1233
